### ISSUE TITLE
Crash when trying to upload photos

### ISSUE THREAD

**Issue by pglevy** [nextcloud/android#13315]
### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
    at com.owncloud.android.ui.adapter.OCFileListAdapter.limitToPersonalFiles(OCFileListAdapter.java:1074)
    at com.owncloud.android.ui.adapter.OCFileListAdapter.swapDirectory(OCFileListAdapter.java:741)
    at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1359)
    at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1308)
    at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1297)
    at com.owncloud.android.ui.activity.FileDisplayActivity.updateListOfFilesFragment(FileDisplayActivity.java:672)
    at com.owncloud.android.ui.activity.FileDisplayActivity$UploadFinishReceiver.onReceive(FileDisplayActivity.java:1382)
    at androidx.localbroadcastmanager.content.LocalBroadcastManager.executePendingBroadcasts(LocalBroadcastManager.java:313)
    at androidx.localbroadcastmanager.content.LocalBroadcastManager$1.handleMessage(LocalBroadcastManager.java:121)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:230)
    at android.os.Looper.loop(Looper.java:319)
    at android.app.ActivityThread.main(ActivityThread.java:8919)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30290290`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a25x`
* Model: `SM-A256U`
* Id: `UP1A.231005.007`
* Product: `a25xsq`

### Firmware
* SDK: `34`
* Release: `14`
* Incremental: `A256USQU3BXE5`

**Comment by alperozturk96** [comment]
@pglevy Could you please try with [3.29.1](https://github.com/nextcloud/android/releases/tag/stable-3.29.1)?

**Comment by joshtrichards** [comment]
Duplicate of #13086
Fixed in #12930
